src/hooks/useUser.spec.js

```diff
@@ -1,17 +1,39 @@
+import React from 'react'
 import useUser from './useUser'
+import IsPublicContext from '../components/IsPublicContext'
 
 import { renderHook } from '@testing-library/react-hooks'
 
+function withPublicContext(value) {
+  // eslint-disable-next-line react/display-name
+  return props => (
+    <IsPublicContext.Provider value={value}>
+      {props.children}
+    </IsPublicContext.Provider>
+  )
+}
+
+function setUrlUsername(username) {
+  const params = username ? 'username=' + encodeURI(username) : ''
+  const url = '/test.html?' + params
+  window.history.pushState({}, 'Test Title', url)
+}
+
 describe('useUser', () => {
   describe('with an explicit username in the url', () => {
     it('should return  this username', () => {
-      window.history.pushState(
-        {},
-        'Test Title',
-        '/test.html?username=hello+world'
-      )
+      setUrlUsername('hello world')
       const { result } = renderHook(() => useUser({}))
       expect(result.current.userName).toEqual('hello world')
     })
   })
+
+  describe('for a public view', () => {
+    it('shows an unknown initial', () => {
+      const context = withPublicContext(true)
+      setUrlUsername(null)
+      const { result } = renderHook(() => useUser({}), { wrapper: context })
+      expect(result.current.userName[0]).toEqual('?')
```

Comment:
why testing only the first char of the string? We should only have one char,no ?

Reply:
Probably, but as the spec from Zoe defines only the initial for the cursor, I wanted the test to reflect the spec and only the spec